solr/core/src/java/org/apache/solr/handler/component/HttpShardHandler.java

```diff
@@ -156,36 +155,31 @@ public void submit(final ShardRequest sreq, final String shard, final Modifiable
       return;
     }
 
-    // all variables that set inside this listener must be at least volatile
-    responseCancellableMap.put(srsp, this.lbClient.asyncReq(lbReq, new AsyncListener<>() {
-      volatile long startTime = System.nanoTime();
-
-      @Override
-      public void onStart() {
-        if (tracer != null && span != null) {
-          tracer.inject(span.context(), Format.Builtin.HTTP_HEADERS, new SolrRequestCarrier(req));
-        }
-        SolrRequestInfo requestInfo = SolrRequestInfo.getRequestInfo();
-        if (requestInfo != null) req.setUserPrincipal(requestInfo.getReq().getUserPrincipal());
-      }
+    long startTime = System.nanoTime();
+    if (tracer != null && span != null) {
```

Comment:
I suspect that this code may be misplaced.  Formerly it was executed by the thread that called onStart.  Is that the same thread calling submit() here?

Reply:
I think they're the same thread? In the previous implementation, the same thread that calls `submit()` [calls the async request method](https://github.com/apache/lucene-solr/blob/master/solr/core/src/java/org/apache/solr/handler/component/HttpShardHandler.java#L160) and then within the async request method [calls `onStart()`](https://github.com/apache/lucene-solr/blob/master/solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java#L89) - so it seems to me like previously the thread that called `submit()` and `onStart()` are the same.